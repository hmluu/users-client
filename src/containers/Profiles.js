import React, { Component } from 'react';

class Profiles extends Component {
  state = {
    isLoading: true,
    profiles: []
  };

  componentDidMount() {
    const API_URL = "http://localhost:3000/api/v1/profiles"
    fetch(API_URL)
      .then(res => res.json())
      .then(profiles => {
        this.setState({
          profiles,
          isLoading: false
        })
      });
  }

  render() {
    return (
      <div>
      <h2>Names</h2>
      {this.state.isLoading ? <h2>Loading Profiles...</h2> : this.state.profiles.map(profile => (
        <li key={profile.id}>{profile.first} {profile.last}</li>
      ))}
      
      </div>
    )
  }
}

export default Profiles;