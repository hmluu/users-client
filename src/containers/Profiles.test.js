import React from 'react';
import ReactDOM from 'react-dom';
import Profiles from './Profiles';

import {shallow, mount} from 'enzyme';

it('renders without crashing', () => {
  shallow(<Profiles/>);
});

it('should render correctly', () => {
  const wrapper = shallow(<Profiles />);
  expect(wrapper).toMatchSnapshot();
});

it('renders without crashing', () => {
  const wrapper = shallow(<Profiles />);
  const title = <h2 className="title">List Of Users</h2>;
  expect(wrapper.contains(title)).toEqual(true);
});

it('should list 3 names', () => {
  const wrapper = shallow(<Profiles/>)
  const list = 3;
  expect(list).toEqual(3);
});

it('renders list-items', () => {
  const profiles = [{
    id: 1,
    first: "John",
    last: "Roberts",
    age: 51,
    location: "Chicago, IL",
    description: "John is a retired YouTuber"
  },
  {
    id: 2,
    first: "Frita",
    last: "Waters",
    age: 27,
    location: "Phoenix, AZ",
    description: "Frita loves the desert, cheese, and turtles."
  },
  {
    id: 3,
    first: "Bart",
    last: "Simpson",
    age: 24,
    location: "Springfield, IL",
    description: "Bart skateboards everywhere."
  }];

  const wrapper = mount(<Profiles profiles={profiles} />);

  // Expect the wrapper object to be defined
  expect(wrapper.find('.users')).toBeDefined();
  expect(wrapper.find('.profiles')).toHaveLength(length);
});

it('Test click event', () => {
  const mockCallBack = jest.fn();

  const click = shallow((<li onClick={mockCallBack}>First Last</li>));
  click.find('li').simulate('click');
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
