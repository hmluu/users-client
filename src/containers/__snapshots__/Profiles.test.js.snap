// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profiles />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="users"
>
          <h2
                    className="title"
          >
                    List Of Users
          </h2>
          <li
                    onClick={[Function]}
          >
                    John
                     
                    Roberts
          </li>
          <li
                    onClick={[Function]}
          >
                    Frita
                     
                    Waters
          </li>
          <li
                    onClick={[Function]}
          >
                    Bart
                     
                    Simpson
          </li>
</ul>,
        <div>
          Select A Profile
</div>,
      ],
      "className": "userList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="title"
>
              List Of Users
</h2>,
            Array [
              <li
                onClick={[Function]}
>
                John
                 
                Roberts
</li>,
              <li
                onClick={[Function]}
>
                Frita
                 
                Waters
</li>,
              <li
                onClick={[Function]}
>
                Bart
                 
                Simpson
</li>,
            ],
          ],
          "className": "users",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "List Of Users",
              "className": "title",
            },
            "ref": null,
            "rendered": "List Of Users",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "John",
                " ",
                "Roberts",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "John",
              " ",
              "Roberts",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Frita",
                " ",
                "Waters",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Frita",
              " ",
              "Waters",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Bart",
                " ",
                "Simpson",
              ],
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Bart",
              " ",
              "Simpson",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select A Profile",
        },
        "ref": null,
        "rendered": "Select A Profile",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="users"
>
            <h2
                        className="title"
            >
                        List Of Users
            </h2>
            <li
                        onClick={[Function]}
            >
                        John
                         
                        Roberts
            </li>
            <li
                        onClick={[Function]}
            >
                        Frita
                         
                        Waters
            </li>
            <li
                        onClick={[Function]}
            >
                        Bart
                         
                        Simpson
            </li>
</ul>,
          <div>
            Select A Profile
</div>,
        ],
        "className": "userList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="title"
>
                List Of Users
</h2>,
              Array [
                <li
                  onClick={[Function]}
>
                  John
                   
                  Roberts
</li>,
                <li
                  onClick={[Function]}
>
                  Frita
                   
                  Waters
</li>,
                <li
                  onClick={[Function]}
>
                  Bart
                   
                  Simpson
</li>,
              ],
            ],
            "className": "users",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "List Of Users",
                "className": "title",
              },
              "ref": null,
              "rendered": "List Of Users",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "John",
                  " ",
                  "Roberts",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "John",
                " ",
                "Roberts",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Frita",
                  " ",
                  "Waters",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Frita",
                " ",
                "Waters",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Bart",
                  " ",
                  "Simpson",
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Bart",
                " ",
                "Simpson",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select A Profile",
          },
          "ref": null,
          "rendered": "Select A Profile",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
